src/Microsoft.AspNetCore.Sockets.Client.Http/SendUtils.cs

```diff
@@ -27,11 +27,12 @@ internal static class SendUtils
                 {
                     // Grab as many messages as we can from the channel
                     messages = new List<SendMessage>();
-                    while (!transportCts.Token.IsCancellationRequested && application.In.TryRead(out SendMessage message))
+                    while (!transportCts.IsCancellationRequested && application.In.TryRead(out SendMessage message))
                     {
                         messages.Add(message);
                     }
 
+                    transportCts.Token.ThrowIfCancellationRequested();
```

Comment:
This is to fix an infinite loop we had

Reply:
What infinite loop? Also accessing the token off the CTS every time isn't great.

Reply:
If the connection is closed but a message was being sent (still in application channel) then we would loop forever. We could instead let `TryRead` happen which will drain the channel and let `httpClient.SendAsync` throw the exception.